src/content/app/tools/blast/state/epics/blastEpics.ts

```diff
@@ -0,0 +1,184 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  from,
+  timer,
+  Subject,
+  pipe,
+  map,
+  mergeMap,
+  concatMap,
+  filter,
+  expand,
+  tap,
+  toArray,
+  EMPTY
+} from 'rxjs';
+import { isFulfilled, type Action } from '@reduxjs/toolkit';
+import type { Epic } from 'redux-observable';
+
+import config from 'config';
+
+import { submitBlast } from '../blast-api/blastApiSlice';
+
+import * as observableApiService from 'src/services/observable-api-service';
+import {
+  saveBlastSubmission,
+  updateSavedBlastJob
+} from 'src/content/app/tools/blast/services/blastStorageService';
+
+import {
+  updateJob,
+  restoreBlastSubmissions,
+  type JobStatus,
+  type BlastJob
+} from 'src/content/app/tools/blast/state/blast-results/blastResultsSlice';
+
+import type { RootState } from 'src/store';
+
+const POLLING_INTERVAL = 15000; // fifteen seconds
+
+/**
+ * An epic that listens to redux action with the data from a BLAST form submission.
+ * It saves the submission to IndexedDB, polls for job statuses,
+ * and when a job is finished or failed, updates its status in indexedDB,
+ * and sends a redux action to update its status in redux
+ */
+export const blastFormSubmissionEpic: Epic<Action, Action, RootState> = (
+  action$
+) =>
+  action$.pipe(
+    filter(submitBlast.matchFulfilled),
```

Comment:
Can we use `ofType` instead of `filter`?

Reply:
I believe `ofType` won't work, because what `redux-toolkit` gives us for RTK-query endpoints is matcher functions for the thunks dispatched by the endpoint rather that a specific action type. As you can see in RTK docs [here](https://redux-toolkit.js.org/rtk-query/api/created-api/endpoints), only `matchFulfilled`, `matchPending` and `matchRejected` functions are available. The `ofType` operator doesn't work with matcher functions :-)

I am a bit skeptical of the whole purpose of the `ofType` operator, to be honest. As you can see from their source code [here](https://github.com/redux-observable/redux-observable/blob/6a6518c1052da17f271ad3a19b28b515d89008e8/src/operators.ts#L16-L48), it's just a filter with some typescript shenanigans, which we already have.